drizzle-zod/src/mysql/index.ts

```diff
@@ -0,0 +1,243 @@
+	PgUUID,
+	PgVarchar,
+} from 'drizzle-orm/mysql-core';
+import { getTableColumns } from 'drizzle-orm/pg-core/utils';
+import { Simplify } from 'drizzle-orm/utils';
+import { z } from 'zod';
+
+type SnakeToCamelCase<S extends string> = S extends `${infer T}${'_' | '-'}${infer U}`
+	? `${T}${Capitalize<SnakeToCamelCase<U>>}`
+	: S;
+
+type ToSnakeCase<S extends string> = S extends `${infer T}${'' | '-'}${infer U}`
+	? `${T extends Capitalize<T> ? Lowercase<T> extends Capitalize<T> ? '' : '_' : ''}${Lowercase<T>}${ToSnakeCase<U>}`
+	: S;
+
+function toCamelCase(value: string): string {
+	return value.replace(/([-_][a-z])/gi, ($1) => $1.toUpperCase().replace('-', '').replace('_', ''));
+}
+
+function toSnakeCase(value: string): string {
+	return value.replace(/([A-Z])/g, ($1) => `_${$1.toLowerCase()}`);
+}
+
+type MaybeOptional<TColumn extends AnyPgColumn, TType extends z.ZodTypeAny, TNoOptional extends boolean = false> =
+	TNoOptional extends true ? TType
+		: GetColumnConfig<TColumn, 'hasDefault'> extends true ? z.ZodOptional<TType>
+		: TType;
+
+type GetZodType<TColumn extends AnyPgColumn> = GetColumnConfig<TColumn, 'data'> extends infer TType
+	? TColumn extends PgEnumColumn<PgEnumColumnConfig & { values: infer TValues extends [string, ...string[]] }>
+		? z.ZodEnum<TValues>
+	: TType extends number ? z.ZodNumber
+	: TType extends string ? z.ZodString
+	: TType extends boolean ? z.ZodBoolean
+	: TType extends Date ? z.ZodDate
+	: z.ZodAny
+	: never;
+
+type ConvertKeyName<K extends string, TCase extends 'snake' | 'camel' | undefined> = undefined extends TCase ? K
+	: 'snake' extends TCase ? ToSnakeCase<K>
+	: SnakeToCamelCase<K>;
+
+export type BuildInsertSchema<
+	TTable extends AnyPgTable,
+	TCase extends 'snake' | 'camel' | undefined,
+	TRefine extends Record<string, z.ZodTypeAny | undefined>,
+	TNoOptional extends boolean = false,
+> = GetTableConfig<TTable, 'columns'> extends infer TColumns extends Record<string, AnyPgColumn> ? {
+		[K in keyof TColumns & string as ConvertKeyName<K, TCase>]: MaybeOptional<
+			TColumns[K],
+			ConvertKeyName<K, TCase> extends keyof TRefine
+				? TRefine[ConvertKeyName<K, TCase>] extends z.ZodTypeAny ? TRefine[ConvertKeyName<K, TCase>]
+				: GetZodType<TColumns[K]>
+				: GetZodType<TColumns[K]>,
+			TNoOptional
+		>;
+	}
+	: never;
+
+export type GetRequiredConfigFields<T extends AnyPgTable> = GetTableConfig<T, 'columns'> extends
+	infer TColumns extends Record<string, AnyPgColumn> ? {
+		[K in keyof TColumns]: TColumns[K] extends PgText<PgTextConfig & { values: infer TValues }>
+			? [string, ...string[]] extends TValues ? never : TValues
+			: never;
+	}
+	: never;
+
+export type RequiredFieldsConfig<
+	TTable extends AnyPgTable,
+> = Simplify<
+	{
+		[K in keyof GetRequiredConfigFields<TTable> as [GetRequiredConfigFields<TTable>[K]] extends [never] ? never : K]: {
+			anyString: true;
+		} | { enum: GetRequiredConfigFields<TTable>[K] };
+	}
+>;
+
+export type Refine<
+	TTable extends AnyPgTable,
```

Comment:
`AnyPgTable` is used for postgres specific functions. We need to use `AnyMySqlTable` here